From ad537f4b68cb00b9d21c60925063c7a060089ea7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Tue, 11 Jul 2023 23:51:10 +0200
Subject: [PATCH] hwdb: drop POINTINGSTICK_CONST_ACCEL
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The TODO says we were supposed to do that in 2019–2020 (if I interpreted the
enigmatic notation correctly). The comment in hwdb said:
> DO NOT USE THIS PROPERTY. This property is kept for backwards
> compatibility. The only known consumer, libinput, stopped reading this
> property in version 1.9.0. No new entries for this property should be
> added.
… and we're currently on libinput-1.23.0.

Most likely there are no users, and even if they are, they'll just get a
slightly misbehaving pointingstick, which shouldn't be too bad.

(cherry picked from commit 28dd29dd95089f676703c7eb355ceb6e2512bb41)

Related: RHEL-5950
---
 TODO                         |  2 --
 hwdb.d/70-pointingstick.hwdb | 44 ------------------------------------
 hwdb.d/parse_hwdb.py         |  1 -
 3 files changed, 47 deletions(-)

diff --git a/TODO b/TODO
index abe8faf5e8..76ce65c529 100644
--- a/TODO
+++ b/TODO
@@ -87,8 +87,6 @@ Deprecations and removals:
 * Remove any support for booting without /usr pre-mounted in the initrd entirely.
   Update INITRD_INTERFACE.md accordingly.
 
-* 2019-10 – Remove POINTINGSTICK_CONST_ACCEL references from the hwdb, see #9573
-
 * remove cgrouspv1 support EOY 2023. As per
   https://lists.freedesktop.org/archives/systemd-devel/2022-July/048120.html
   and then rework cgroupsv2 support around fds, i.e. keep one fd per active
diff --git a/hwdb.d/70-pointingstick.hwdb b/hwdb.d/70-pointingstick.hwdb
index a78d088173..6ff228bee6 100644
--- a/hwdb.d/70-pointingstick.hwdb
+++ b/hwdb.d/70-pointingstick.hwdb
@@ -51,28 +51,11 @@
 #
 # Allowed properties are:
 #   ID_INPUT_POINTINGSTICK
-#   POINTINGSTICK_CONST_ACCEL (deprecated)
 #   POINTINGSTICK_SENSITIVITY
 #
 # Entries should be sorted with growing _SENSITIVITY and _CONST_ACCEL.
 #
 #########################################
-#      POINTINGSTICK_CONST_ACCEL        #
-#########################################
-#
-# DO NOT USE THIS PROPERTY. This property is kept for backwards
-# compatibility. The only known consumer, libinput, stopped reading this
-# property in version 1.9.0. No new entries for this property should be
-# added.
-#
-# Trackpoint const accel settings are specified as
-#    POINTINGSTICK_CONST_ACCEL=<accel>
-#
-# Where <accel> is a floating point number, using a '.' separator, specifying
-# by how much to multiply deltas generated by the pointingstick to get
-# normalized deltas.
-#
-#########################################
 #      POINTINGSTICK_SENSITIVITY        #
 #########################################
 #
@@ -93,30 +76,6 @@ evdev:name:*[tT]rack[pP]oint*:*
 evdev:name:*[dD]ual[pP]oint [sS]tick*:*
  ID_INPUT_POINTINGSTICK=1
 
-#########################################
-# Dell
-#########################################
-
-# Latitude D620
-evdev:name:*DualPoint Stick:dmi:bvn*:bvr*:bd*:svnDellInc.:pnLatitudeD620*:*
- POINTINGSTICK_CONST_ACCEL=0.5
-
-# Latitude E5570
-evdev:name:*DualPoint Stick:dmi:bvn*:bvr*:bd*:svnDellInc.:pnLatitudeE5570*:*
- POINTINGSTICK_CONST_ACCEL=0.1
-
-# Latitude E6320
-evdev:name:*DualPoint Stick:dmi:bvn*:bvr*:bd*:svnDellInc.:pnLatitudeE6320*:*
- POINTINGSTICK_CONST_ACCEL=2.0
-
-# Latitude E6400
-evdev:name:*DualPoint Stick:dmi:bvn*:bvr*:bd*:svnDellInc.:pnLatitudeE6400*:*
- POINTINGSTICK_CONST_ACCEL=1.5
-
-# Latitude E7470
-evdev:name:*DualPoint Stick:dmi:bvn*:bvr*:bd*:svnDellInc.:pnLatitudeE7470*:*
- POINTINGSTICK_CONST_ACCEL=0.6
-
 #########################################
 # Lenovo
 #########################################
@@ -157,7 +116,6 @@ evdev:name:TPPS/2 IBM TrackPoint:dmi:bvn*:bvr*:bd*:svnLENOVO:pn*:pvrThinkPadX1Ca
 # Lenovo Thinkpad X1 Tablet
 evdev:name:TPPS/2 IBM TrackPoint:dmi:bvn*:bvr*:bd*:svnLENOVO:pn*:pvrThinkPadX1Tablet:*
  POINTINGSTICK_SENSITIVITY=200
- POINTINGSTICK_CONST_ACCEL=1.0
 
 # Lenovo Thinkpad X200/X201/X200s/X201s/X200 Tablet/X201 Tablet
 # Note these come with 2 revisions of keyboard, with the trackpoints having a
@@ -167,9 +125,7 @@ evdev:name:TPPS/2 IBM TrackPoint:dmi:bvn*:bvr*:bd*:svnLENOVO:pn*:pvrThinkPadX20?
 evdev:name:TPPS/2 IBM TrackPoint:dmi:bvn*:bvr*:bd*:svnLENOVO:pn*:pvrThinkPadX20??:*
 evdev:name:TPPS/2 IBM TrackPoint:dmi:bvn*:bvr*:bd*:svnLENOVO:pn*:pvrThinkPadX20?Tablet:*
  POINTINGSTICK_SENSITIVITY=200
- POINTINGSTICK_CONST_ACCEL=1.25
 
 # Lenovo UltraNav SK-8845 (USB keyboard)
 evdev:input:b0003v06CBp0009*
- POINTINGSTICK_CONST_ACCEL=2.5
  POINTINGSTICK_SENSITIVITY=200
diff --git a/hwdb.d/parse_hwdb.py b/hwdb.d/parse_hwdb.py
index ee3474a6f5..30d5f8a569 100755
--- a/hwdb.d/parse_hwdb.py
+++ b/hwdb.d/parse_hwdb.py
@@ -174,7 +174,6 @@ def property_grammar():
              ('ID_INPUT_TRACKBALL', Or((Literal('0'), Literal('1')))),
              ('ID_SIGNAL_ANALYZER', Or((Literal('0'), Literal('1')))),
              ('POINTINGSTICK_SENSITIVITY', INTEGER),
-             ('POINTINGSTICK_CONST_ACCEL', REAL),
              ('ID_INPUT_JOYSTICK_INTEGRATION', Or(('internal', 'external'))),
              ('ID_INPUT_TOUCHPAD_INTEGRATION', Or(('internal', 'external'))),
              ('XKB_FIXED_LAYOUT', xkb_setting),
